python/openmlconverter/src/converter.py

```diff
@@ -0,0 +1,151 @@
+import datetime
+from functools import partial
+
+import dateutil.parser
+import langcodes
+
+
+def bake(openml_dataset: dict, openml_features: dict) -> dict:
+    _ds = partial(_get_field, json_dict=openml_dataset)  # get field from openml_dataset
+
+    distributions = [
+        _distribution(name=_ds(field="name"), url=url)
+        for url in sorted({_ds(field="minio_url"), _ds(field="url"), _ds(field="parquet_url")})
+    ]
+    distribution_source = distributions[0]["name"]
+    croissant = {
+        "@context": {"@vocab": "https://schema.org/", "ml": "http://mlcommons.org/schema/"},
+        "@type": "Dataset",
+        "@language": "en",
+        "name": _ds(field="name", required=True),
+        "description": _ds(field="description", required=True),
+        "version": _ds(field="version"),
+        "creator": _ds(
+            field="creator",
+            transform=lambda v: [_person(p) for p in v] if isinstance(v, list) else _person(v),
+        ),
+        "contributor": _ds(field="contributor", transform=_person),
+        "dateCreated": _ds(field="upload_date", transform=dateutil.parser.parse),
+        "dateModified": _ds(field="processing_date", transform=dateutil.parser.parse),
+        "datePublished": _ds(field="collection_date", transform=_lenient_date_parser),
+        "inLanguage": _ds(field="language", transform=lambda v: langcodes.find(v).language),
+        "isAccessibleForFree": True,
+        "license": _ds(field="license"),
+        "creativeWorkStatus": _ds(field="status"),
+        "keywords": _ds(field="tag"),
+        "citation": _ds(field="citation"),
+        "sameAs": _ds(field="original_data_url"),
+        "url": f"https://www.openml.org/api/v1/json/data/{_ds(field='id')}",
+        "distribution": distributions,
+        "recordSet": [
+            {
+                "name": _ds(field="name"),
+                "@type": "ml:RecordSet",
+                "source": f"#{{{distribution_source}}}",
+                "key": _row_identifier(openml_features),
+                "field": [
+                    {
+                        "name": feat["name"],
+                        "@type": "ml:Field",
+                        "dataType": _datatype(feat["data_type"], feat.get("nominal_value", None)),
+                        "source": f"#{{{distribution_source}/{feat['name']}}}",
+                    }
+                    for feat in openml_features
+                ],
+            }
+        ],
+    }
+    _remove_none_values(croissant)
+    for recordSet in croissant["recordSet"]:
+        _remove_none_values(recordSet)
+    return croissant
+
+
+def _get_field(json_dict, field: str, transform=None, required=False):
+    """
+    Convenience function. Get field and optionally perform the transformation. If field does not
+    exist, throw an error if the field is  required, otherwise return None.
```

Comment:
Please document the function arguments and return value as well.

Reply:
At Google this is the docstring formatting we use: https://github.com/google/styleguide/blob/gh-pages/pyguide.md#38-comments-and-docstrings

I have no strong feelings about using one or another format, as long as we are consistent within each sub-project.

Reply:
Thanks! I used the Google docstring formatting.
For this file: https://github.com/mlcommons/datasets_format/pull/3/commits/a83b64ba1f9fe2beec1bac970fd0354c4c10d94b
For other files: https://github.com/mlcommons/datasets_format/pull/3/commits/40e89fe17fed52629b9d235dd14f7dfde2c59751